index.js

```diff
@@ -20,18 +20,8 @@ module.exports.isFinished = isFinished
  * @private
  */
 
-var asyncHooks = tryRequireAsyncHooks()
-var first = require('ee-first')
-
-/**
- * Variables.
- * @private
- */
-
-/* istanbul ignore next */
-var defer = typeof setImmediate === 'function'
-  ? setImmediate
-  : function (fn) { process.nextTick(fn.bind.apply(fn, arguments)) }
+const asyncHooks = require('node:async_hooks')
+const stream = require('node:stream')
```

Comment:
```suggestion
const asyncHooks = require('async_hooks')
const stream = require('stream')
```
Depending on the version support range we want to land on here, `node:foo` didn't go stable available until node 16 I think?

Reply:
@jonchurch What version range do we want to support here? Since this change requires a new major release anyway, we could take the opportunity to bump the range to ">=18" to match what Express is using.

Reply:
cc @wesleytodd  @UlisesGascon

I don't have a principled opinion here right now. The project as a whole may not either.

18 would be pragmatic, but we can get away with 10.

For context: the main reason this package is still useful IMO is that it combines the monitoring of sockets along with streams to determine when a req/res has "finished". I believe there are still some edgecases where a socket can close and the req/res would miss the close event. Hence the need to solve in libraryland vs relying solely on the platform (aka `stream.finished`).

Because the change here is minor (nobody really _needs_ this feature alone) targetting 18 has low risk of leaving someone behind.